src/main/java/org/spongepowered/common/data/processor/value/entity/FireDamageDelayValueProcessor.java

```diff
@@ -0,0 +1,84 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.value.entity;
+
+import static org.spongepowered.common.data.util.ComparatorUtil.intComparator;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.MutableBoundedValue;
+import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+import org.spongepowered.common.data.value.mutable.SpongeBoundedValue;
+
+public class FireDamageDelayValueProcessor extends AbstractSpongeValueProcessor<Integer, MutableBoundedValue<Integer>> {
+
+    public FireDamageDelayValueProcessor() {
+        super(Keys.FIRE_DAMAGE_DELAY);
+    }
+
+    @Override
+    public Optional<Integer> getValueFromContainer(ValueContainer<?> container) {
+        if (supports(container)) {
+            return Optional.of(((net.minecraft.entity.Entity) container).fireResistance);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public boolean supports(ValueContainer<?> container) {
+        return container instanceof net.minecraft.entity.Entity;
+    }
+
+    @Override
+    public DataTransactionResult offerToStore(ValueContainer<?> container, Integer value) {
+        final ImmutableValue<Integer> proposedValue = new ImmutableSpongeValue<Integer>(Keys.FIRE_DAMAGE_DELAY, value);
+        if (supports(container)) {
+            final ImmutableValue<Integer> newFireDelayData = new ImmutableSpongeValue<Integer>(Keys.FIRE_DAMAGE_DELAY, value);
+            final ImmutableValue<Integer> oldFireDelayValue = getApiValueFromContainer(container).get().asImmutable();
+            try {
+                ((net.minecraft.entity.Entity) (container)).fireResistance = value;
+                return DataTransactionBuilder.successReplaceResult(oldFireDelayValue, newFireDelayData);
+            } catch (Exception e) {
```

Comment:
Don't catch an exception here- the only thing that can throw one is `DataTransactionBuilder`, which shouldn't happen (and therefore shouldn't be suppressed).